### PR TITLE
Generate and integrate the REST API in the Reference Manual

### PR DESCRIPTION
**Pull request by FabienLelaquais**
- Generate REST API from scratch
- Remove Node.JS stuff
- GUI sources have moved to src/taipy

### CODE REVIEW

File: docs/manuals/reference_rest/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Taipy's REST API
```
Comment (jrobinAV) [inline docs/manuals/reference_rest/index.md:1]: This file is related to ref_man and not user_man. I would not put this file under `manuals` folder, but I would create another folder `reference`. 
The index.md file path would be `docs/reference/rest/index.md`.

Does it make sense to you?

**Review by jrobinAV — APPROVED** [review]
Great.

As soon as it is merged, I will work on improving the content itself.

Thx!

File: docs/manuals/reference_rest/index.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/manuals/reference_rest/index.md:1]: It does. This should be the 'reference_rest' directory.
Cannot use 'reference', which is 100 % generated...
Thanks!

Comment (FabienLelaquais) [reply docs/manuals/reference_rest/index.md:1]: Oh. I was going to fix this.... but I don't need to.
We do agree: this file is specific to the Reference REST API Manual. And in my opinion, located at the right place.
Do you agree with that?

**Pull request merged**